reader.go

```diff
@@ -31,17 +33,28 @@ func Decode(r io.Reader) (image.Image, error) {
 		read += 4
 		switch string(next) {
 		case "TOC ":
-			read += 4
+			resSize := binary.BigEndian.Uint32(data[read : read+4])
+			read += resSize-4 // size includes header and size fields
 			continue
 		case "icnV":
 			read += 4
 			continue
 		}
+
+		resSize := binary.BigEndian.Uint32(data[read : read+4])
+		read += 4
+		if resSize == 0 {
+			continue // no content, we're not interested
+		}
+
+		iconData := data[read : read+resSize-8]
+		read += resSize-8 // size includes header and size fields
```

Comment:
Similarly here, I think a more specific name like `dataSize` would be slightly better (instead of `resSize`), because "data" is already in the nomenclature, for example with `iconData`.